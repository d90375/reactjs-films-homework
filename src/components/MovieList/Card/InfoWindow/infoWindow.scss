.infoWindow {
  position: absolute;
  top: 0;
  width: 100%;
  height: 100%;
  border-radius: 6px;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  overflow: hidden;
}

.infoWindow__btn {
  outline: none;
  border: none;
  background-color: transparent;
  color: #fff;
  font-size: 2rem;
  padding: 0;
  margin: 0.5rem 0.8rem;
  cursor: pointer;

  &:hover {
    color: darken(#fff, 20%);
  }
}

.infoWindow__wrap {
  padding: 0 1rem 0 1rem;
  box-sizing: border-box;
  width: 100%;
}

.infoWindow__title {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between;
}

.infoWindow__title_text {
  font-family: "Helvetica Neue - bold", serif;
  font-size: 18px;
  font-weight: 700;
  text-transform: uppercase;
  letter-spacing: 1.2px;
  width: 14rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0;
  line-height: 22px;
  flex-wrap: nowrap;
  color: #fff;
}

.infoWindow__title_score {
  margin: 0;
  font-family: "Helvetica Neue - Condensed Black", serif;
  font-size: 18px;
  width: 30px;
  height: 19px;
  font-weight: 500;
  color: #47c3f0;
  text-align: center;
  border-radius: 5px;
  border: 1px solid #49c1f0;
}

.infoWindow__genres {
  margin: 0;
  padding-top: 0.2rem;
  font-family: "Helvetica Neue - bold", serif;
  font-size: 0.7rem;
  line-height: 16px;
  letter-spacing: 0.88px;
  font-weight: 500;
  color: #49c1f0;
}

.infoWindow_overview {
  margin: 0;
  padding: 1rem;
  color: #fff;
  font-family: "Helvetica Neue - thin", serif;
  font-size: 1rem;
  line-height: 1.6rem;
  letter-spacing: 1px;
  overflow: hidden;
  text-overflow: clip;
}

.infoWindow_btnWrap {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  margin: 1rem 0 1rem 1rem;
}
