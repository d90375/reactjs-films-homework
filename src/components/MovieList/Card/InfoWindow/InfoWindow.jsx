import React from "react";
import PropTypes from "prop-types";
import ButtonWatchNow from "../../../VideoFrame/ButtonWatchNow";

import { IMAGE_NOT_FOUND_URL } from "../../../../constants";
import "./infoWindow.scss";

const InfoWindow = ({ cardItem, onShowInfo }) => {
  const { title, score, genres, posterImg, overview, id } = cardItem;
  const infoWindowBackgroundStyle = {
    background: posterImg
      ? `linear-gradient(0deg, rgba(0, 0, 0, 0.7) 100%, rgba(0, 0, 0, 0.7) 100%),
    url(https://image.tmdb.org/t/p/w500${posterImg})`
      : `url(${IMAGE_NOT_FOUND_URL}) 0% 0% / contain`,
    backgroundSize: posterImg ? "cover" : "contain",
    backgroundRepeat: "no-repeat",
    backgroundPosition: "top"
  };

  return (
    <div className="infoWindow" style={infoWindowBackgroundStyle}>
      <button className="infoWindow__btn" type="button" onClick={onShowInfo}>
        &#215;
      </button>
      <div className="infoWindow__wrap">
        <div className="infoWindow__title">
          <h3 className="infoWindow__title_text">{title}</h3>
          <p className="infoWindow__title_score">{score}</p>
        </div>
        <p className="infoWindow__genres">{genres}</p>
      </div>
      <p className="infoWindow_overview">{overview}</p>
      <div className="infoWindow_btnWrap">
        <ButtonWatchNow movieId={id} />
      </div>
    </div>
  );
};

export default InfoWindow;

InfoWindow.propTypes = {
  cardItem: PropTypes.shape({
    genres: PropTypes.string,
    score: PropTypes.number,
    title: PropTypes.string,
    posterImg: PropTypes.string,
    overview: PropTypes.string,
    id: PropTypes.number
  }),
  onShowInfo: PropTypes.func.isRequired
};

InfoWindow.defaultProps = {
  cardItem: {
    genres: "",
    score: 0,
    title: "",
    posterImg: "",
    overview: "",
    id: 0
  }
};
