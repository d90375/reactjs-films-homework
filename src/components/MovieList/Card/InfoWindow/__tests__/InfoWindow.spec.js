import React from "react";
import InfoWindow from "../InfoWindow";

const setUp = (props) => {
  return shallow(<InfoWindow {...props} />);
};

describe("InfoWindow component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  const mockProps = {
    cardData: {
      genres: "genres",
      score: 999,
      title: "title",
      posterImg: "posterImg",
      overview: "overview",
      id: 9
    }
  };

  it("should render InfoWindow component without props", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render InfoWindow component with props", () => {
    component = setUp({ ...mockProps });
    expect(component).toMatchSnapshot();
  });

  describe("defaultProps", () => {
    it("should use default onShowInfo", () => {
      const result = InfoWindow.defaultProps.onShowInfo();
      expect(result).toBe(undefined);
    });
  });
});
