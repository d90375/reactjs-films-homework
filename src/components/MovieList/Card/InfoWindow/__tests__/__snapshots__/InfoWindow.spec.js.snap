// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoWindow component should render InfoWindow component with props 1`] = `
<div
  className="infoWindow"
  style={
    Object {
      "background": "linear-gradient(0deg, rgba(0, 0, 0, 0.7) 100%, rgba(0, 0, 0, 0.7) 100%),
    url(https://image.tmdb.org/t/p/w500posterImg)",
      "backgroundPosition": "top",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
    }
  }
>
  <button
    className="infoWindow__btn"
    onClick={[Function]}
    type="button"
  >
    ×
  </button>
  <div
    className="infoWindow__wrap"
  >
    <div
      className="infoWindow__title"
    >
      <h3
        className="infoWindow__title_text"
      >
        title
      </h3>
      <p
        className="infoWindow__title_score"
      >
        999
      </p>
    </div>
    <p
      className="infoWindow__genres"
    >
      genres
    </p>
  </div>
  <p
    className="infoWindow_overview"
  >
    overview
  </p>
  <div
    className="infoWindow_btnWrap"
  >
    <Memo(ButtonWatchNow)
      movieId={9}
    />
  </div>
</div>
`;

exports[`InfoWindow component should render InfoWindow component without props 1`] = `
<div
  className="infoWindow"
  style={
    Object {
      "background": "url(https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-38-picture-grey-c2ebdbb057f2a7614185931650f8cee23fa137b93812ccb132b9df511df1cfac.svg) 0% 0% / contain",
      "backgroundPosition": "top",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "contain",
    }
  }
>
  <button
    className="infoWindow__btn"
    onClick={[Function]}
    type="button"
  >
    ×
  </button>
  <div
    className="infoWindow__wrap"
  >
    <div
      className="infoWindow__title"
    >
      <h3
        className="infoWindow__title_text"
      />
      <p
        className="infoWindow__title_score"
      >
        0
      </p>
    </div>
    <p
      className="infoWindow__genres"
    />
  </div>
  <p
    className="infoWindow_overview"
  />
  <div
    className="infoWindow_btnWrap"
  >
    <Memo(ButtonWatchNow)
      movieId={0}
    />
  </div>
</div>
`;
